daemon/daemon_unix.go

```diff
@@ -623,6 +636,51 @@ func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.
 	return warnings, nil
 }
 
+func (daemon *Daemon) loadRuntimes() error {
+	return daemon.initRuntimes(daemon.configStore.Runtimes)
+}
+
+func (daemon *Daemon) initRuntimes(runtimes map[string]types.Runtime) (err error) {
+	runtimeDir := filepath.Join(daemon.configStore.Root, "runtimes")
+	tmpDir := runtimeDir + fmt.Sprintf("-%d", time.Now().UTC().UnixNano())
+	if err = os.MkdirAll(tmpDir, 0700); err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			if err = os.Rename(runtimeDir, runtimeDir+"-old"); err != nil {
+				return
+			}
+			if err = os.Rename(tmpDir, runtimeDir); err != nil {
+				err = errors.Wrapf(err, "failed to setup runtimes directory, new containers may not start!")
+				return
+			}
+			if err = os.RemoveAll(runtimeDir + "-old"); err != nil {
+				logrus.WithError(err).WithField("dir", tmpDir).
+					Warnf("failed to remove old runtimes dir")
+			}
+		} else {
```

Comment:
Maybe we can `if err != nil` at the top and return early from this defer and no need for `else`